basic.py

```diff
@@ -90,3 +91,33 @@ def eval_(event):
         return eval(input, {"__builtins__": None}, safe_dict)
     except Exception as e:
         return "eval: {0}".format(e)
+
+
+last_events = {}
+
+
+@event
+def sub_event_catcher(event):
+    if event.event_type in ['message', 'cmd']:
+        last_events[event.meta['host']] = event
+
+
+@match(regex=r'^s/(.*?)/(.*?)/([a-z]*)$')
+def sub_match(event, pattern, repl, flags, *args, **kwargs):
```

Comment:
`repl` reminds me more of `REPL` than `replacement`. I wonder if there isn't a better name for that argument.